### PR TITLE
Deal with MiniSEED header text decoding issues

### LINKED ISSUE #2177: Skip invalid unicode in MiniSEED headers
The Gecko data logger (https://gecko.kelunji.net/) seems to write invalid unicode into some miniseed headers [(Example file)](https://github.com/obspy/obspy/files/2116776/2018-05-29131100GECKO.ms.zip). The error message is attached at the end of the post.

My proposed solution would be to add the errors='ignore' argument to the decode()-calls in 
https://github.com/obspy/obspy/blob/master/obspy/io/mseed/util.py#L631
and following
```python
info["station"] = data[:5].strip().decode()
info["location"] = data[5:7].strip().decode()
info["channel"] = data[7:10].strip().decode()
info["network"] = data[10:12].strip().decode()
```

as well as
https://github.com/obspy/obspy/blob/master/obspy/io/mseed/core.py#L473

Python version: 2.7 or 3.5
Obspy version: 1.1.0 (installed via anaconda)

```
In [2]: st = obspy.read('2018-05-29131100GECKO.ms')
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-2-27d0c93fb1f1> in <module>()
----> 1 st = obspy.read('2018-05-29131100GECKO.ms')

<decorator-gen-146> in read(pathname_or_url, format, headonly, starttime, endtime, nearest_sample, dtype, apply_calib, check_compression, **kwargs)

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/core/util/decorator.py in _map_example_filename(func, *args, **kwargs)
    299                         except IOError:
    300                             pass
--> 301         return func(*args, **kwargs)
    302     return _map_example_filename
    303 

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/core/stream.py in read(pathname_or_url, format, headonly, starttime, endtime, nearest_sample, dtype, apply_calib, check_compression, **kwargs)
    233         pathname = pathname_or_url
    234         for file in sorted(glob(pathname)):
--> 235             st.extend(_read(file, format, headonly, **kwargs).traces)
    236         if len(st) == 0:
    237             # try to give more specific information why the stream is empty

<decorator-gen-147> in _read(filename, format, headonly, **kwargs)

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/core/util/decorator.py in uncompress_file(func, filename, *args, **kwargs)
    209     else:
    210         # no compressions
--> 211         result = func(filename, *args, **kwargs)
    212     return result
    213 

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/core/stream.py in _read(filename, format, headonly, **kwargs)
    275     """
    276     stream, format = _read_from_plugin('waveform', filename, format=format,
--> 277                                        headonly=headonly, **kwargs)
    278     # set _format identifier for each element
    279     for trace in stream:

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/core/util/base.py in _read_from_plugin(plugin_type, filename, format, **kwargs)
    410         raise TypeError(msg % (format_ep.name, ', '.join(eps)))
    411     # read
--> 412     list_obj = read_format(filename, **kwargs)
    413     return list_obj, format_ep.name
    414 

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/io/mseed/core.py in _read_mseed(mseed_object, starttime, endtime, headonly, sourcename, reclen, details, header_byteorder, verbose, **kwargs)
    281         raise ObsPyMSEEDFilesizeTooLargeError(msg)
    282 
--> 283     info = util.get_record_information(mseed_object, endian=bo)
    284 
    285     # Map the encoding to a readable string value.

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/io/mseed/util.py in get_record_information(file_or_file_object, offset, endian)
    540     if isinstance(file_or_file_object, (str, native_str)):
    541         with open(file_or_file_object, 'rb') as f:
--> 542             info = _get_record_information(f, offset=offset, endian=endian)
    543     else:
    544         info = _get_record_information(file_or_file_object, offset=offset,

~/anaconda3/envs/Py35/lib/python3.5/site-packages/obspy/io/mseed/util.py in _get_record_information(file_object, offset, endian)
    628     data = file_object.read(12)
    629     info["station"] = data[:5].strip().decode()
--> 630     info["location"] = data[5:7].strip().decode()
    631     info["channel"] = data[7:10].strip().decode()
    632     info["network"] = data[10:12].strip().decode()

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf0 in position 0: invalid continuation byte
```

### PR DESCRIPTION
**Pull request by krischer**
Fixes #2177 

It will first attempt to correctly decode it as ASCII (thus not more ambiguity based on the system encoding) and if that fails it will just ignore the offending characters and raise a hopefully useful warning.

I feel like this is a good compromise.

CC: @sstaehler

### CODE REVIEW

File: obspy/io/mseed/scripts/recordanalyzer.py [hunk 1]
```diff
@@ -175,17 +175,22 @@ def _read_fixed_header(self):
             raise
         # Write values to dictionary.
         self.fixed_header['Sequence number'] = \
-            int(''.join(x.decode('ascii') for x in header_item[:6]))
+            int(''.join(x.decode('ascii', errors="warn")
```
Comment (megies) [inline obspy/io/mseed/scripts/recordanalyzer.py]: Does `.decode(..., errors="warn")` exist? I don't see it.. https://docs.python.org/3.4/library/codecs.html#error-handlers

**Comment by megies** [comment]
As of now, the record analyzer won't show a warning when encountering such invalid files, or am I missing something?

**Comment by krischer** [comment]
Yea i'll change that.

**Comment by sstaehler** [comment]
looks good from my side and fixes the problem

**Comment by krischer** [comment]
I used `.decode(..., errors="replace")` everywhere in the record analyzer. Given that this is mainly intended as a debugging tool I think this is the best choice so one can see where things go wrong.

**Pull request merged**